### PR TITLE
[MM-57743] Enable errcheck linter

### PR DESCRIPTION
**Pull request by hanzei**
#### Summary
The PR enables the `errcheck` linter, something that was planned for a few years now. 

The ignore list is quite broad. We can scope it down more granularly once we have a clear idea of how we do that. (Campain, staff member, don't work on it for now?)

First, I want to fix the issues in the enterprise code, as we can't do that with the community.

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-57743

#### Release Note
```release-note
NONE
```

### CODE REVIEW

File: server/.golangci.yml [hunk 1]
```diff
@@ -27,10 +27,10 @@ linters:
     - makezero
     - whitespace
     # TODO: enable this later
-    # - errcheck
+    - errcheck
 
 issues:
-  skip-dirs:
+  exclude-dirs:
```
Comment (hanzei) [inline server/.golangci.yml:32]: `skip-dirs` doesn't exist. Fixing that as part of this PR.

File: server/.golangci.yml [hunk 2]
```diff
@@ -26,11 +26,10 @@ linters:
     - goimports
     - makezero
     - whitespace
-    # TODO: enable this later
```
Comment (hanzei) [inline server/.golangci.yml:29]: 4 years later, we can finally remove this

File: server/config/utils.go [hunk 3]
```diff
@@ -104,14 +104,11 @@ func fixConfig(cfg *model.Config) {
...
+	fixInvalidLocales(cfg)
 }
 
-// FixInvalidLocales checks and corrects the given config for invalid locale-related settings.
-//
-// Ideally, this function would be completely internal, but it's currently exposed to allow the cli
-// to test the config change before allowing the save.
-func FixInvalidLocales(cfg *model.Config) bool {
+// fixInvalidLocales checks and corrects the given config for invalid locale-related settings.
+func fixInvalidLocales(cfg *model.Config) bool {
```
Comment (hanzei) [inline server/config/utils.go:111]: Fixing the comment while touching the package

File: server/config/utils_test.go [hunk 4]
```diff
@@ -80,7 +80,9 @@ func TestDesanitize(t *testing.T) {
 }
 
 func TestFixInvalidLocales(t *testing.T) {
-	utils.TranslationsPreInit()
+	// utils.TranslationsPreInit errors when TestFixInvalidLocales is run as part of testing the package,
+	// but doesn't error when the test is run individually.
+	_ = utils.TranslationsPreInit()
```
Comment (hanzei) [inline server/config/utils_test.go:85]: I haven't found out why, though.

File: server/.golangci.yml [hunk 5]
```diff
@@ -57,3 +56,7 @@ issues:
     - linters:
       - staticcheck
       text: SA1019
+
+    - linters:
+        - errcheck
+      path: "channels|platform|cmd/mattermost|public/model|public/plugin|public/shared/mlog|enterprise/data_retention|enterprise/saml|enterprise/cloud|enterprise/ldap"
```
Comment (hanzei) [inline server/.golangci.yml:62]: The ignore list is quite broad. We can scope it down more granularly once we have a clear idea of how we do that. (Campain, staff member, don't work on it for now?)

First, I want to fix the issues in the enterprise code, as we can't do that with the community.

**Review by agnivade — APPROVED** [review]

File: server/config/utils_test.go (hunk 4 above)
Comment (isacikgoz) [reply server/config/utils_test.go:85]: Do we know the error? Maybe we can create a ticket to track or create a HW issue.

**Review by isacikgoz — APPROVED** [review]
LGTM

**Pull request merged**